toppra/dracula/__init__.py

```diff
@@ -236,14 +236,15 @@ def compute_spline_varying_alim(self):
         for multiplier in t_sum_multipliers:
             path = self._estimate_path(multiplier, pc_vel, pc_acc)
             if (self.qlim is not None):
-                for _ in range(MAX_RESPLINES):
+                while self.resplines_allowed > 0:
                     if self.joint_limits_obeyed(path, multiplier):
                         break
                     logger.info(
                         f"Path violates joint limits. Re-estimating."
                     )
                     logger.debug(f"waypts = {self.waypts}")
                     path = self._estimate_path(multiplier, pc_vel, pc_acc)
+                    self.resplines_allowed -= 1
```

Comment:
how does the re-estimating differ from the initial estimate above in L237? they seem to be the same call using the same input params

Reply:
If the spline violates joint limits, then `self.joint_limits_obeyed` adds extra knot points to `self.waypts` which is used to respline in `self._estimate_path`.  So yes the calls are identical but the state of the caller has changed between calls.

Reply:
ok please put a comment there stating `self.waypts` gets modified when false. this is unclear to readers unless there's comment

Reply:
I put in a comment for now.  Since I'm working to get us switched to the other version of toppra, I don't want to spend too much time maintaining this version.  If the new version of toppra doesn't work well, I'll re-engineer how the joint limits are checked/waypoints are updated later this week.